File: android/src/main/kotlin/com/itoken/team/wutspider/interfaces/ISpider.kt
package com.itoken.team.wutspider.interfaces

import com.itoken.team.wutspider.SpiderException
import com.itoken.team.wutspider.model.SpiderInfo

interface ISpider {

    fun serialize(info: SpiderInfo): String

    fun deserialize(info: String): SpiderInfo

    fun createClient(info: SpiderInfo): ISpiderHttpClient

    @Throws(SpiderException::class)
    fun run(
        info: SpiderInfo,
        environment: Map<String, String>,
        client: ISpiderHttpClient? = null
    ): Map<String, String>

}

File: dotnet/SpiderEngine/Abstract/ISpider.cs
namespace SpiderEngine.Abstract;

using SpiderEngine.Model;

public interface ISpider
{
    ISpiderHttpClient CreateClient(SpiderInfo spiderInfo);
    IDictionary<string, string> Run(SpiderInfo spiderInfo, IDictionary<string, string> environment, ISpiderHttpClient? client = null);
    SpiderInfo Deserialize(string json);
    string Serialize(SpiderInfo spiderInfo);
    string GetSchema();
}
